herd/ASLSem.ml

```diff
@@ -311,33 +308,42 @@ module Make (C : Config) = struct
           return @@ shortcut b v
       | _ -> M.op herdop v1 v2
 
+    let concat v1 v2 =
+      match (v1, v2) with
+      | V.Val (Constant.Concrete (ASLScalar.S_BitVector bv)), v
+        when Asllib.Bitvector.length bv = 0 ->
+          return v
+      | v, V.Val (Constant.Concrete (ASLScalar.S_BitVector bv))
+        when Asllib.Bitvector.length bv = 0 ->
+          return v
+      | _ -> M.op (Op.ArchOp ASLOp.Concat) v1 v2
```

Comment:
Hi @HadrienRenaud. Why this change that looks unrelated to this PR purpose?

Reply:
Why not! But why so? Maybe this is worth a comment?

Reply:
Perfectly so. In fact I was wondering why the shortcut is here and  not in ASLOp. In ant case, this is nitpicking.

Reply:
C'est trop juste.

Reply:
Seriously, such optimisations cannot be performed in `lib/SymbValue.ml` because, at this stage, we have no idea of the semantics of `ArchOp` operations. Not sure this is worth fixing...